src/ng/parse.js

```diff
@@ -1788,6 +1797,7 @@ function $ParseProvider() {
             }
             cache[cacheKey] = parsedExpression;
           }
+          parsedExpression = expensiveChecksInterceptor(parsedExpression, expensiveChecks);
```

Comment:
Also, wouldn't it make sense to cache the `parsedExpression` that has already had the expensive checks interceptor added, if the cache is `cacheExpensive`?